shell/common/skp_shader_warmup_unittests.cc

```diff
@@ -0,0 +1,237 @@
+  shell->GetTaskRunners().GetIOTaskRunner()->PostTask(
+      [&io_task_finished]() { io_task_finished.set_value(true); });
+  io_task_finished.get_future().wait();
+}
+
+class SkpWarmupTest : public ShellTest {
+ public:
+  SkpWarmupTest(){}
+
+  void TestWarmup(const SkISize &draw_size, const LayerTreeBuilder &builder){
+    // Create a temp dir to store the persistent cache
+    fml::ScopedTemporaryDirectory dir;
+    PersistentCache::SetCacheDirectoryPath(dir.path());
+    PersistentCache::ResetCacheForProcess();
+
+    auto settings = CreateSettingsForFixture();
+    settings.cache_sksl = true;
+    settings.dump_skp_on_shader_compilation = true;
+
+    fml::AutoResetWaitableEvent firstFrameLatch;
+    settings.frame_rasterized_callback =
+        [&firstFrameLatch](const FrameTiming& t) { firstFrameLatch.Signal(); };
+
+    auto config = RunConfiguration::InferFromSettings(settings);
+    config.SetEntrypoint("emptyMain");
+    std::unique_ptr<Shell> shell = CreateShell(settings);
+    PlatformViewNotifyCreated(shell.get());
+    RunEngine(shell.get(), std::move(config));
+
+    // Initially, we should have no SkSL cache
+    auto cache = PersistentCache::GetCacheForProcess()->LoadSkSLs();
+    ASSERT_EQ(cache.size(), 0u);
+
+
+    PumpOneFrame(shell.get(), draw_size.width(), draw_size.height(), builder);
+    firstFrameLatch.Wait();
+    WaitForIO(shell.get());
+
+    // Count the number of shaders this builder generated. We use this as a proxy
+    // for whether new shaders were generated, since skia will dump an skp any
+    // time a new shader is compiled.
+    int skp_count = 0;
+    fml::FileVisitor skp_count_visitor = [&skp_count](const fml::UniqueFD& directory,
+                                                const std::string& filename) {
+      if (filename.size() >= 4 &&
+          filename.substr(filename.size() - 4, 4) == ".skp") {
+        skp_count += 1;
+      }
+      return true;
+    };
+    fml::VisitFilesRecursively(dir.fd(), skp_count_visitor);
+    int first_skp_count = skp_count;
+    skp_count = 0;
+    ASSERT_GT(first_skp_count, 0);
+
+    // Deserialize all skps into memory
+    std::vector<sk_sp<SkPicture>> pictures;
+    fml::FileVisitor skp_deserialize_visitor = [&pictures](const fml::UniqueFD& directory,
+                                                const std::string& filename) {
+      if (filename.size() >= 4 &&
+          filename.substr(filename.size() - 4, 4) == ".skp") {
+        auto fd = fml::OpenFileReadOnly(directory, filename.c_str());
+        if(fd.get() < 0){
+          FML_LOG(ERROR) << "Failed to open " << filename;
+          return true;
+        }
+        // Deserialize 
+        sk_sp<SkData> data = SkData::MakeFromFD(fd.get());
+        std::unique_ptr<SkMemoryStream> stream = SkMemoryStream::Make(data);
+
+        SkDeserialProcs procs = {0};
+        procs.fImageProc = DeserializeImageWithoutData;
+        sk_sp<SkPicture> picture = SkPicture::MakeFromStream(stream.get(), &procs);
```

Comment:
I'm a little surprised that `SkPicture::MakeFromStream` can be used here without changing Skia's build configuration. I think Flutter has disabled that Skia feature. Is there a Fuchsia config file that re-enables it?

BTW, I can't find `SkpWarmupTest` in https://api.cirrus-ci.com/v1/task/6670703166488576/logs/test_host.log

Reply:
First question i dunno, seems to work for me.

Second question: the whole test is ifdef'd with OS_FUCHSIA right now because it was failing on windows and i figured it was best to isolate any problems with it to Fuchsia since thats the only platform where this is relevant right now. I assume that means its not included in host_debug_unopt